builder/chroot-script.sh

```diff
@@ -1,51 +1,167 @@
+
+  if [[ ${#GPG_KEY} -eq 16 ]]; then
+    echo "compare long keyid"
+    CHECK=$(clean_print long "${FINGERPRINT_OF_FILE}")
+  elif [[ ${#GPG_KEY} -eq 8 ]]; then
+    echo "compare short keyid"
+    CHECK=$(clean_print short "${FINGERPRINT_OF_FILE}")
+  else
+    echo "compare fingerprint"
+    CHECK=$(clean_print fpr "${FINGERPRINT_OF_FILE}")
+  fi
+
+  if [[ "${GPG_KEY}" == "${CHECK}" ]]; then
+    echo "key OK add to apt"
+    apt-key add "${KEY_FILE}"
+    rm -f "${KEY_FILE}"
+    return 0
+  else
+    echo "key invalid"
+    exit 1
+  fi
+}
+
+
+## examples:
+# clean_print {print|fpr|long|short} {GPGKEYID|FINGERPRINT}
+# get_gpg {GPGKEYID|FINGERPRINT} [URL|FILE]
+
 # device specific settings
 HYPRIOT_DEVICE="ODROID C2"
 
 # set up /etc/resolv.conf
-echo "nameserver 8.8.8.8" > /etc/resolv.conf
+DEST=$(readlink -m /etc/resolv.conf)
+export DEST
+mkdir -p "$(dirname "${DEST}")"
+echo "nameserver 8.8.8.8" > "${DEST}"
 
 # set up ODROID repository
-apt-key adv --keyserver keyserver.ubuntu.com --recv-keys AB19BAC9
+ODROID_KEY_ID=AB19BAC9
+get_gpg $ODROID_KEY_ID
 echo "deb http://deb.odroid.in/c2/ xenial main" > /etc/apt/sources.list.d/odroid.list
 
-# # set up Hypriot Schatzkiste repository
-# wget -q https://packagecloud.io/gpg.key -O - | apt-key add -
-# echo 'deb https://packagecloud.io/Hypriot/Schatzkiste/debian/ wheezy main' > /etc/apt/sources.list.d/hypriot.list
+# set up hypriot arm repository for odroid packages
+PACKAGECLOUD_FPR=418A7F2FB0E1E6E7EABF6FE8C2E73424D59097AB
+PACKAGECLOUD_KEY_URL=https://packagecloud.io/gpg.key
+get_gpg "${PACKAGECLOUD_FPR}" "${PACKAGECLOUD_KEY_URL}"
+
+# set up hypriot schatzkiste repository for generic packages
+echo 'deb https://packagecloud.io/Hypriot/Schatzkiste/debian/ jessie main' >> /etc/apt/sources.list.d/hypriot.list
+
+# add armhf as additional architecure (see below)
+dpkg --add-architecture armhf
 
 # update all apt repository lists
 export DEBIAN_FRONTEND=noninteractive
 apt-get update
+apt-get upgrade -y
+
+# define packages to install
+packages=(
+    # as the Odroid C2 does not have a hardware clock we need a fake one
+    fake-hwclock
+
+    # install device-init
+    device-init:armhf=${DEVICE_INIT_VERSION}
+
+    # install dependencies for docker-tools
+    lxc
+    aufs-tools
+    cgroupfs-mount
+    cgroup-bin
+    apparmor
+    libltdl7
+
+    # install docker-compose, and docker-machine
+    libc6:armhf
+    zlib1g:armhf
+    docker-compose:armhf=${DOCKER_COMPOSE_VERSION}
+    docker-machine:armhf=${DOCKER_MACHINE_VERSION}
+)
+
+apt-get -y install ${packages[*]}
+
+# install linux kernel for Odroid C2
+apt-get -y install u-boot-tools linux-image-c2
+
+# install docker-engine
+DOCKER_DEB=$(mktemp)
+wget -q -O "$DOCKER_DEB" "$DOCKER_DEB_URL"
+echo "${DOCKER_DEB_CHECKSUM} ${DOCKER_DEB}" | sha256sum -c -
+dpkg -i "$DOCKER_DEB"
+
+# Restore os-release additions
+cat /tmp/os-release.add >> /etc/os-release
```

Comment:
well, yes and no. The problem this is fixing will always reappear when a new version of base-files gets releases since os-rootfs was. This will result in failed test runs. I'll happily revert this if you think it's not worth the effort, but it took me a while to figure out why the tests were failing.